doc/developer_guide/psyir.rst

```diff
@@ -521,7 +521,42 @@ sections of the reference guide.
         datatype of the operands (e.g. in the select-case canonicalisation).
         A solution to this is to create an abstract interface with appropriate
         implementations for each possible datatype.
-            
+
+Data Type of an Operation Node
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+For unary operations, the type of the result is just that of
+the operand.
```

Comment:
Given that this are also part of table 7.2, it feels like this sentence should be part the paragraph below, maybe move it to before "For a numeric, binary operation..."

Reply:
Oh yes, that's better, thanks.